### PR TITLE
Check if id_lang exist before delete

### LINKED ISSUE #27506: Unknown column 'id_lang' in 'where clause'
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add screenshots

When i tried to delete an language. 
When a table finished by `_lang`

PrestaShop goes inside and try to remove the entry by the le key `id_lang`.

if this field doesn't exist, the error is triggered

Capture d’écran 2022-01-28 à 11 21 47

### Expected behavior

Delete the language Any way

### Steps to reproduce

Go to your backoffice International > Translations. 
And a language. 

Create this table for testing: 
```sql
CREATE TABLE `ps_test_lang` (
  `id_test` int(10) unsigned NOT NULL,
  `lang` int(10) unsigned NOT NULL,
  PRIMARY KEY (`id_test`,`lang`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```

And try to delete the language added. 

### PrestaShop version(s) where the bug happened

1.7.8.3

### PHP version(s) where the bug happened

_No response_

### If your bug is related to a module, specify its name and its version

_No response_

### PR DESCRIPTION
**Pull request by PrestaSafe**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | fix of issue https://github.com/PrestaShop/PrestaShop/issues/27506
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/27506
| How to test?      | see issue
| Possible impacts? | None

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/27507)

### CODE REVIEW

**Comment by PrestaSafe** [comment]
Update table: 
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | fix of issue https://github.com/PrestaShop/PrestaShop/issues/27506
| Type?             | bug fix
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes https://github.com/PrestaShop/PrestaShop/issues/27506
| How to test?      | see issue
| Possible impacts? | None

File: classes/Language.php [hunk 1]
```diff
@@ -607,8 +607,11 @@ public function delete()
 
             foreach ($result as $row) {
                 if (isset($row[$tableNameKey]) && !empty($row[$tableNameKey]) && preg_match('/_lang$/', $row[$tableNameKey])) {
-                    if (!Db::getInstance()->execute('DELETE FROM `' . $row[$tableNameKey] . '` WHERE `id_lang` = ' . (int) $this->id)) {
-                        return false;
+                    $columns =  Db::getInstance()->executeS('SHOW COLUMNS FROM `' . $row[$tableNameKey] . '` LIKE "id_lang" ');
```
Comment (PululuK) [inline classes/Language.php]:
```suggestion
                    $columns = Db::getInstance()->executeS('SHOW COLUMNS FROM `' . $row[$tableNameKey] . '` LIKE "id_lang" ');
```

**Review by PululuK — DISMISSED** [review]
LGTM !
Thanks @PrestaSafe :)

**Review by PululuK — APPROVED** [review]

**Review by PrestaEdit — CHANGES_REQUESTED** [review]
As mention on the issue itself, I'm not sure that we need this.

Imho, creating a _lang table without id_lang and/or the _shop table without id_shop it's already a creation bug.

**Comment by PrestaSafe** [comment]
I agree with you but for the final customer it can be complicated at the end. 

So a few more checking step is welcome 😬

**Review by matks — APPROVED** [review]
I think it's a good thing to always check that what you want to erase exists beforehands 👍  good idea @PrestaSafe

**Review by florine2623 — CHANGES_REQUESTED** [review]
Hello @PrestaSafe ,

Thanks for the PR !

I have some feedback.
I'm encountering an exception after testing your PR, here are the steps I followed :
- Added multiple languages in International > Translations
- Create this table for testing:
```
CREATE TABLE `ps_test_lang` (
  `id_test` int(10) unsigned NOT NULL,
  `lang` int(10) unsigned NOT NULL,
  PRIMARY KEY (`id_test`,`lang`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8;
```
- Delete an installed language OK
- Try to add another language  from International > Translations NOK 🚫 
Screenshot 2022-02-08 at 09 40 30

Could you check please ?
Thanks!

**Comment by Progi1984** [comment]
Hi @PrestaSafe,
Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.
Thanks!

**Pull request closed without merging**